### PR TITLE
Add sit out functionality to tablemenu.

### PR DESCRIPTION
**Pull request by mtutynina**

### CODE REVIEW

File: js/table/tablemanager.ts [hunk 1]
```diff
@@ -501,6 +502,15 @@ export class TableManager {
 
         this.currentIndex(0);
     }
+    public toggleSkipDeals(value: boolean) {
```
Comment (kant2002) [inline js/table/tablemanager.ts]: This functionality should not be part of the `TableManager`, since TableManager responsible to keep track of the TableView, This functionality should belongs to the TableView, and I suppose that could be implemented using `TableViwe.sitOut()` and `TableView.comeBack()` methods.

File: js/table/actionBlock.ts [hunk 2]
```diff
@@ -554,13 +555,7 @@ export class ActionBlock {
...
             }
 
-            const gameApi = new Game(host);
-            if (value) {
-                gameApi.sitOut(this.tableView.tableId);
-            } else {
-                gameApi.comeBack(this.tableView.tableId);
-            }
-
+            tableManager.toggleSkipDeals(value);
```
Comment (kant2002) [inline js/table/actionBlock.ts]: This call imply that call to server would be part of `ActionBlock` which prevent it from running in the test suite. Or we have to add some unnescessary mocking. You should add such subscription to the TableView utilize KO subscription. For example: `this.actionBlock.skipDeals.subscribe((value) => this.toggleSkipDeals(value))`.

In general you should view ActionBlock as dummy which just represent UI, and all actions done by TableView which control ActionBlock and subscribe to it notifications, to do actual work.

**Review by kant2002 — CHANGES_REQUESTED** [review]

File: js/table/tableview.ts [hunk 3]
```diff
@@ -2309,6 +2317,15 @@ export class TableView {
             self.reportApiError(data.Status);
         }
     }
+
+    public toggleSkipDeals(value: boolean) {
```
Comment (kant2002) [inline js/table/tableview.ts]: Name parameter as `skipDeals`

File: js/table/actionBlock.ts [hunk 4]
```diff
@@ -210,10 +210,11 @@ export class ActionBlock {
      */
     public gameFinished: KnockoutObservable<boolean>;
 
+    public suppressSetSitoutStatus: boolean;
```
Comment (kant2002) [inline js/table/actionBlock.ts]: Why this change? Looks like this variable should belongs to TableView now ? Am I miss some other place where it is used?

**Review by kant2002 — CHANGES_REQUESTED** [review]

File: js/table/actionBlock.ts (hunk 4 above)
Comment (mtutynina) [reply js/table/actionBlock.ts]: line number 469, should i move this function to tableview ?

Comment (kant2002) [reply js/table/actionBlock.ts]: If it don't used anywhere in the `ActionBlock`, then please move it to `TableView`

File: js/table/tableview.ts [hunk 5]
```diff
@@ -584,6 +594,15 @@ export class TableView {
             return (currentPlayer.Status() & 1) !== 0;
         }, this);
 
+        this.isSitOut.subscribe((value) => {
```
Comment (kant2002) [inline js/table/tableview.ts]: Should this be `this.actionBlock.isSitOut` ?

Comment (kant2002) [reply js/table/tableview.ts]: Looks like whole moving of `suppressSetSitoutStatus` is unnescessary, since it is sould belongs to ActionBlock and it should be private, since this is implementation details.

Comment (mtutynina) [reply js/table/tableview.ts]: No

Comment (kant2002) [reply js/table/tableview.ts]: I see that this variable related somehow to the fact that we have panel work in sync with `skipDeals` checkbox.

File: js/popups/tablemenupopup.ts [hunk 6]
```diff
@@ -334,6 +340,12 @@ export class TableMenuPopup {
             leaved.then(removeCurrentTable);
         }
     }
+    private toggleSkipDeals() {
+        this.skipDeals(!this.skipDeals());
+        const currentIndex = tableManager.currentIndex();
+        const tableView = tableManager.tables()[currentIndex];
```
Comment (kant2002) [inline js/popups/tablemenupopup.ts]: Please use `const currentTable = this.currentTableProvider.currentTable();` for getting reference to current table

File: js/popups/tablemenupopup.ts [hunk 7]
```diff
@@ -334,6 +340,12 @@ export class TableMenuPopup {
...
+        tableView.toggleSkipDeals(this.skipDeals());
```
Comment (kant2002) [inline js/popups/tablemenupopup.ts]: It looks like ActionBlock is already have such functionality and `tableView.actionBlock.skipDeals(this.skipDeals())` do the actual trick. At the same time, the current subscription on the 
https://github.com/online-poker/poker-html-client/blob/3e94a7093f97061cdfe049cb427e56e165614660/js/table/actionBlock.ts#L552
indicate that we have code small in that place. Action block should not know about API calls, so you could hide relationship to API by calling `this.tableView.toggleSkipDeals(value)` there.

That way we could at least move subscription later.

**Review by kant2002 — CHANGES_REQUESTED** [review]

File: js/popups/tablemenupopup.ts [hunk 8]
```diff
@@ -147,6 +152,7 @@ export class TableMenuPopup {
         this.handHistoryAllowed(playerIsInGame && currentTable.lastHandHistory() != null);
         this.leaveAllowed(myPlayer != null && !currentTable.myPlayerInGame() && myPlayer.IsSitoutStatus());
         this.accountStatusAllowed(this.authInformation.authenticated());
+        this.skipDeals(currentTable.isSitOut());
```
Comment (kant2002) [inline js/popups/tablemenupopup.ts:155]: nitpicking: Could it be `currentTable.skipDeals()` for consistency?

Comment (mtutynina) [reply js/popups/tablemenupopup.ts:155]: we don't have skipDeals on tableview

**Review by kant2002 — DISMISSED** [review]

**Review by kant2002 — APPROVED** [review]
